aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+
+[!code-javascript[Configuring logging in the JavaScript client](diagnostics/logging-config-js.js?highlight=3)]
+
+> [!NOTE]
+> To disable logging entirely, specify `signalR.LogLevel.None` in the `configureLogging` method.
+
+Log levels available to the JavaScript client are listed below. Setting the log level to one of these values enables logging of message at that level, as well as all log levels above it on the table below:
+
+| Level | Description |
+| ----- | ----------- |
+| `None` | No messages are logged. |
+| `Critical` | Messages that indicate a failure in the entire app. |
+| `Error` | Messages that indicate a failure in the current operation. |
+| `Warning` | Messages that indicate a non-fatal problem. |
+| `Information` | Informational messages. |
+| `Debug` | Diagnostic messages useful for debugging. |
+| `Trace` | Very detailed diagnostic messages designed for diagnosing specific issues. |
+
+Once you've configured the verbosity, the logs will be written to the Browser Console (or Standard Output in a NodeJS application).
+
+If you want to send logs to a custom logging system, you can provide a JavaScript object implementing the `ILogger` interface. The only method that needs to be implemented is `log`, which takes the level of the event and the message associated with the event. For example:
+
+[!code-typescript[Creating a custom logger](diagnostics/custom-logger.ts?highlight=3-7,13)]
+
+## .NET client logging
+
+> [!NOTE]
+> Client-side logs may contain sensitive information from your application. **Never** post raw logs from production applications to public forums like GitHub.
+
+To get logs from the .NET client, you can use the `.ConfigureLogging` method on `HubConnectionBuilder`. This works the same way as the `.ConfigureLogging` method on `WebHostBuilder` and `HostBuilder`. You can configure the same logging providers you use in ASP.NET Core. However, you have to manually install and enable the NuGet packages for the individual logging providers.
+
+### Console logging
+
+In order to enable Console logging, add the `Microsoft.Extensions.Logging.Console` nuget package. Then, use the `AddConsole` method to configure the console logger:
+
+[!code-csharp[Configuring console logging in .NET client](diagnostics/net-client-console-log.cs?highlight=6)]
+
+### Debug output window logging
+
+You can also configure logs to go to the Output Window in Visual Studio. Install the `Microsoft.Extensions.Logging.Debug` package and use the `AddDebug` method:
+
+[!code-csharp[Configuring debug output window logging in .NET client](diagnostics/net-client-debug-log.cs?highlight=6)]
+
+### Other logging providers
+
+You can also use other logging providers such as Serilog, Seq, NLog, or any other logging system that integrates with `Microsoft.Extensions.Logging`. If your logging system provides an `ILoggerProvider`, you can register it with `.AddProvider`:
```

Comment:
Nit: The above three sections start with "You can also" which is a little repetitive